### DISCUSSION TITLE
6.0 issues with Publishing tab and uploading into Geoserver

### DISCUSSION THREAD

**Discussion by jenw-ua** [ua-gist604b-f24/syllabus#25] [General]
Did uploading the .sld file into Geoserver work for anyone?? I also get a white screen of death when I click the Publishing tab, so after many attempts of that, I skipped that part and moved forward. Now I've been stuck at a standstill trying to upload the .sld file into Geoserver. It will not upload in any way. I can copy and paste it, but it will not validate, save, or apply. I've killed codespace and redone the assignment multiple times...for two work days now...here are my last errors. Anybody know what to do or anybody else stuck in this boat?

<img width="2551" alt="Screenshot 2024-11-19 at 7 50 39 PM" src="https://github.com/user-attachments/assets/d54f6909-1e94-4a1c-8603-719438a36c79">

**Comment by wtb4rtr** [comment]
Yes I am having the same problem; I have tried uploading the style about 15 or 20 times and nothing ever happens; then I pasted the xml code for the file and tried to Validate, Apply, Save, nothing works.  I don't get any error messages or any  messages saying its processing just no response at all from geoserver

**Comment by travisz09** [comment]
I also had lots of issues with my .sld file. I was able to upload my file fine, but every time I tried to validate I got a Javascript error which I could not decipher.  I ended up loading a .sld file from one of the existing styles in geoserver and compared it to my QGIS output. There seemed to be many differences in the xml tags, and the existing style used version 1.0.0, while QGIS used version 1.1.0. I do not know if that is significant, or not.

Anyway, I ended up just working around this by using one of the existing styles as a template and making the changes I wanted to the shape, fill, and stroke tags until it matched my QGIS style. This allowed me to move forward, but obviously was not an ideal solution. Taking a peek ahead at assignment 6-1, I see that .sld files are used repeatedly, so hopefully someone else has a better solution that they can share.

Best regards,
Travis

Edit: I just went back into geoserver and tried my QGIS .sld again. This is the error.

```
java.lang.RuntimeException: Parsing failed for Graphic: java.lang.ClassCastException: class java.lang.String cannot be cast to class org.geotools.api.filter.expression.Expression (java.lang.String is in module java.base of loader 'bootstrap'; org.geotools.api.filter.expression.Expression is in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @62a8fd44)
```

**Reply by jenw-ua** [reply]
Wow. Mine are still dead at trying to upload in any kind of way on any computer on any browser. I was having java errors too when I looking in the developer window. Can't resolve them with my available resources though :-/

**Reply by aaryno** [reply]
@travisz09 I googled your specific error and found this suggesting it was an error parsing the SLD file: https://gis.stackexchange.com/questions/397464/geoserver-sld-parsing-error-on-unknown-rule 

We would probably have to diagnose that with your specific SLD which of course will be different than other students' SLD files. In any case, we are definitely looking at a case where one tool (QGIS) is likely producing valid SLD but another tool (geoserver) is not parsing it correctly. It could be the other way around (QGIS producing invalid SLD) but I'm guessing not. QGIS is a vibrant, supported community and geoserver is languishing. I'll discuss this in a lecture update.

**Reply by kev-bone** [reply]
I don't recall the specific error, but I had trouble uploading SLD files with more complex symbology. I added a gradient fill to the land use polygon in 6-1 for example, and geoserver wouldn't validate that. Once I changed it back to a simple fill it uploaded fine.

**Comment by aldunn4** [comment]
I only got it to work using Safari on my iPad. Neither Chrome nor Edge worked on my PC. Apple discontinued Safari for Windows, so I put my GeoServer URL and sld file in an email to myself, downloaded it to the iPad and took my screenshot for Q3 there and sent it back to myself to continue working on my Windows PC.

**Reply by jenw-ua** [reply]
Wow, that's nuts.  It looks like this module should have been a group project and maybe we could've completed it, especially considering the MANY hours we've all spent on this module.  Good luck with the rest of class! Thanks for the feedback.

**Comment by travisz09** [comment]
@jenw-ua, are you still stuck on this? Do you want to give it another try?

**Reply by aaryno** [reply]
I'll be sitting in https://arizona.zoom.us/s/86236244627 if you are working on this now.
